File: lib/py/test/thryft_test/core/protocol/test/_protocol_test.py
#-------------------------------------------------------------------------------
# Copyright (c) 2013, Minor Gordon
# All rights reserved.
# 
# Redistribution and use in source and binary forms, with or without
# modification, are permitted provided that the following conditions
# are met:
# 
#     * Redistributions of source code must retain the above copyright
#       notice, this list of conditions and the following disclaimer.
# 
#     * Redistributions in binary form must reproduce the above copyright
#       notice, this list of conditions and the following disclaimer in
#       the documentation and/or other materials provided with the
#       distribution.
# 
# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
# CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
# INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
# MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL  DAMAGES (INCLUDING,
# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
# OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
# OF SUCH DAMAGE.
#-------------------------------------------------------------------------------

from decimal import Decimal
from thryft_test.protocol.test.protocol_test_enum import ProtocolTestEnum
from thryft_test.protocol.test.protocol_test_struct import ProtocolTestStruct
import unittest


class _ProtocolTest(unittest.TestCase):
    def test_bool(self):
        self._test(ProtocolTestStruct(bool_field=True))

    def test_byte(self):
        self._test(ProtocolTestStruct(byte_field=1))

#    def test_date(self):
#        self._test(ProtocolTestStruct.Builder().setDateField(DateTime.now())
#                .build());
#
#    def test_dateTime(self):
#        self._test(ProtocolTestStruct.Builder()
#                .setDateTimeField(DateTime.now()).build());

    def test_decimal(self):
        self._test(ProtocolTestStruct(decimal_field=Decimal(100)))

    def test_enum(self):
        self._test(ProtocolTestStruct(enum_field=ProtocolTestEnum.ENUMERATOR2))

    def test_i16(self):
        self._test(ProtocolTestStruct(i16_field=1))

    def test_i32(self):
        self._test(ProtocolTestStruct(i32_field=1))

    def test_i64(self):
        self._test(ProtocolTestStruct(i64_field=1l))

    def test_list_string(self):
        self._test(ProtocolTestStruct(list_string_field=('test1', 'test2')))

    def test_map_string_string(self):
        self._test(ProtocolTestStruct(map_string_string_field={'testkey': 'testvalue', 'testkey2': 'testvalue2'}))

    def test_set_string(self):
        self._test(ProtocolTestStruct(set_string_field=frozenset(('test1', 'test2'))))

    def test_string(self):
        self._test(ProtocolTestStruct(string_field='test'))

    def test_Struct(self):
        self._test(ProtocolTestStruct(struct_field=ProtocolTestStruct()))

    def _test(self, in_object):
        pass

File: lib/java/protocol/src/test/java/org/thryft/protocol/test/ProtocolTest.java
/*******************************************************************************
 * Copyright (c) 2013, Minor Gordon
 * All rights reserved.
 * 
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions
 * are met:
 * 
 *     * Redistributions of source code must retain the above copyright
 *       notice, this list of conditions and the following disclaimer.
 * 
 *     * Redistributions in binary form must reproduce the above copyright
 *       notice, this list of conditions and the following disclaimer in
 *       the documentation and/or other materials provided with the
 *       distribution.
 * 
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
 * CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
 * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
 * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
 * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
 * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
 * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL  DAMAGES (INCLUDING,
 * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
 * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
 * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
 * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
 * OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
 * OF SUCH DAMAGE.
 ******************************************************************************/

package org.thryft.protocol.test;

import java.math.BigDecimal;

import org.joda.time.DateTime;
import org.junit.Test;
import org.thryft.native_.EmailAddress;
import org.thryft.native_.Url;

import com.google.common.collect.ImmutableList;
import com.google.common.collect.ImmutableMap;
import com.google.common.collect.ImmutableSet;

public abstract class ProtocolTest {
    @Test
    public void testBool() throws Exception {
        _test(new ProtocolTestStruct.Builder().setBoolField(true));
    }

    @Test
    public void testByte() throws Exception {
        _test(new ProtocolTestStruct.Builder().setByteField((byte) 1));
    }

    @Test
    public void testDateTime() throws Exception {
        _test(new ProtocolTestStruct.Builder().setDateTimeField(DateTime.now()));
    }

    @Test
    public void testDecimal() throws Exception {
        _test(new ProtocolTestStruct.Builder().setDecimalField(new BigDecimal(
                100)));
    }

    @Test
    public void testEmailAddress() throws Exception {
        _test(new ProtocolTestStruct.Builder()
                .setEmailAddressField(new EmailAddress("test@example.com")));
    }

    @Test
    public void testEnum() throws Exception {
        _test(new ProtocolTestStruct.Builder()
                .setEnumField(ProtocolTestEnum.ENUMERATOR2));
    }

    @Test
    public void testI16() throws Exception {
        _test(new ProtocolTestStruct.Builder().setI16Field((short) 1));
    }

    @Test
    public void testI32() throws Exception {
        _test(new ProtocolTestStruct.Builder().setI32Field(1));
    }

    @Test
    public void testI64() throws Exception {
        _test(new ProtocolTestStruct.Builder().setI64Field(1));
    }

    @Test
    public void testListString() throws Exception {
        _test(new ProtocolTestStruct.Builder().setListStringField(ImmutableList
                .of("test")));

        // Empty list
        _test(new ProtocolTestStruct.Builder().setListStringField(ImmutableList
                .<String> of()));
    }

    @Test
    public void testMapStringString() throws Exception {
        _test(new ProtocolTestStruct.Builder()
                .setMapStringStringField(ImmutableMap
                        .of("testkey", "testvalue")));

        // Empty map
        _test(new ProtocolTestStruct.Builder()
                .setMapStringStringField(ImmutableMap.<String, String> of()));
    }

    @Test
    public void testSetString() throws Exception {
        _test(new ProtocolTestStruct.Builder().setSetStringField(ImmutableSet
                .of("test")));

        // Empty set
        _test(new ProtocolTestStruct.Builder().setSetStringField(ImmutableSet
                .<String> of()));
    }

    @Test
    public void testString() throws Exception {
        _test(new ProtocolTestStruct.Builder().setStringField("test"));
    }

    @Test
    public void testStruct() throws Exception {
        _test(new ProtocolTestStruct.Builder()
                .setStructField(new ProtocolTestStruct.Builder().setI32Field(1)
                        .setRequiredI32Field(1).setRequiredStringField("test")
                        .build()));

        // Empty struct
        _test(new ProtocolTestStruct.Builder()
                .setStructField(new ProtocolTestStruct.Builder()
                        .setRequiredI32Field(1).setRequiredStringField("test")
                        .build()));
    }

    @Test
    public void testUrl() throws Exception {
        _test(new ProtocolTestStruct.Builder().setUrlField(Url
                .parse("http://example.com/test")));
    }

    protected abstract void _test(final ProtocolTestStruct expected)
            throws Exception;

    protected void _test(final ProtocolTestStruct.Builder expectedBuilder)
            throws Exception {
        _test(expectedBuilder.setRequiredI32Field(1)
                .setRequiredStringField("test").build());
    }
}
